src/encoded/static/components/cart/analysis.js

```diff
@@ -0,0 +1,207 @@
+import _ from 'underscore';
+import { computeAssemblyAnnotationValue } from '../objectutils';
+
+
+/** Lab path we see for the ENCODE uniform processing pipeline */
+const UNIFORM_PIPELINE_LAB = '/labs/encode-processing-pipeline/';
+
+/** Order that all possible lab strings should sort. */
+const labSortOrder = [
+    'ENCODE4',
+    'ENCODE3',
+    'Uniform',
+    'Lab custom',
+];
+
+
+/**
+ * Compile the usable experiment analysis objects into a form for rendering a dropdown of pipeline
+ * labs.
+ * @param {object} experiment Contains the analyses to convert into an pipeline labs dropdown
+ * @param {array} files Array of all files from all datasets in the cart.
+ *
+ * @return {array} Compiled analyses information, each element with the form:
+ * {
+ *      title: Analysis dropdown title
+ *      status: Analysis object submission status
+ *      lab: Generally ENCODE or other
+ *      version: Pipeline version number -- "major.minor.patch"
+ *      assembly: Assembly string matching assembly facet terms
+ *      annotation: Genome annotation string
+ *      assemblyAnnotationValue: Value used to sort assembly/annotation
+ *      files: Array of files selected with the pipeline lab and assembly
+ * }
+ */
+const compileAnalysesByTitle = (experiment, files) => {
```

Comment:
Good idea. compileAnalysis in filegallery is a bit unstable as tweaks to features are happening in release week. Maybe in the future this can be refactored and merged with filegallery's version